scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/instances/SparkeyWriter.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.extra.sparkey.instances
+
+import com.spotify.scio.extra.sparkey.SparkeyUri
+import com.spotify.scio.util.RemoteFileUtil
+import com.spotify.sparkey.{CompressionType, Sparkey}
+
+import java.io.File
+import java.net.URI
+import java.nio.file.{Files, Paths}
+
+private[sparkey] class SparkeyWriter(
+  val uri: SparkeyUri,
+  rfu: RemoteFileUtil,
+  compressionType: CompressionType,
+  compressionBlockSize: Int,
+  maxMemoryUsage: Long = -1
+) {
+  private val localFile =
```

Comment:
can we make this lazy so it's not executed until the pipeline is run? I'm just concerned as-is it wouldn't work with PTransformOverride, for example